test/constraint/encoding.test.ts

```diff
@@ -445,27 +445,197 @@ describe('constraints/encoding', () => {
   });
 
   describe('scalePropertiesSupportedByScaleType', () => {
-    let encQ: EncodingQuery = {
-      channel: Channel.X,
-      field: 'A',
-      type: Type.QUANTITATIVE,
-      scale: {type: undefined, zero: true}
-    };
+    describe('scaleBandSize', () => {
+      let encQ: EncodingQuery = {
+        channel: Channel.X,
+        field: 'A',
+        type: Type.QUANTITATIVE,
+        scale: {bandSize: 10, type: undefined}
+      };
 
-    it('should return false if scaleType does not support scaleZero', () => {
-      [ScaleType.LOG, ScaleType.ORDINAL, ScaleType.TIME, ScaleType.UTC].forEach((scaleType) => {
-        (encQ.scale as ScaleQuery).type = scaleType;
-        assert.isFalse(ENCODING_CONSTRAINT_INDEX['scalePropertiesSupportedByScaleType'].satisfy(encQ, schema, defaultOpt));
+      it('should return false if scaleTYpe does not suport scaleBandSize', () => {
```

Comment:
You never test for a case where scale type is `undefined` and we infer type from scale type.   
(You don't have to add this for all properties, but at least should test it once.) 

Please utilize the test coverage tool to see if you coverage all **branches** (not just statement) for `scalePropertiesSupportedByScaleType` in `src/constraint/encoding.ts`